src/lightning/data/streaming/combined.py

```diff
@@ -0,0 +1,93 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+from typing import Any, Dict, Iterator, List, Optional, Sequence
+
+from torch.utils.data import IterableDataset
+
+from lightning.data.streaming.dataset import StreamingDataset
+
+
+class CombinedStreamingDataset(IterableDataset):
```

Comment:
Can you bring over the tests from lit-gpt?

Reply:
done !